python/ray/air/util/tensor_extensions/arrow.py

```diff
@@ -488,7 +488,10 @@ def _concat_same_type(
                 [e for a in to_concat for e in a]
             )
         else:
-            storage = pa.concat_arrays([c.storage for c in to_concat])
+            if len(to_concat) == 1:
+                storage = to_concat[0].storage
```

Comment:
can we add a comment for why this is needed, for reader in the future.

Reply:
It turns out that the PyArrow Block copy code path expects this function to copy data. I reverted this change in this PR, as it doesn't impact the final benchmark perf.